runtime-api/bonded-currency/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use parity_scale_codec::{alloc::string::String, Codec};
+use sp_std::vec::Vec;
+
+mod pool_details;
+pub use pool_details::*;
+
+sp_api::decl_runtime_apis! {
+	/// Runtime API to compute the collateral for a given amount and pool ID
+	/// and to query all pool IDs where the given account is the manager or owner.
+	#[api_version(1)]
+	pub trait BondedCurrency<Balance, PoolId, Operation, AccountId, BitType, PoolDetails, Error> where
+		Balance: Codec,
+		PoolId: Codec,
+		Operation: Codec,
+		AccountId: Codec,
+		BitType: Codec,
+		PoolDetails: Codec,
+		Error: Codec
+		{
+			/// Calculates the collateral for the given amount.
+			/// The operation is determining whether the amount is minted or burned.
+			/// The calculated collateral amount is based on the current state of the pool.
+			fn get_quota(
```

Comment:
What about a more descriptive name than `quota`?

Reply:
It was previously `calculate_collateral_for_amount` raphael complained and said `get_quota` is a better name. What about `get_collateral`?

Reply:
That sounds clearer to me.

Reply:
[e12f907](https://github.com/KILTprotocol/kilt-node/pull/828/commits/e12f907916bfe519fc2bce7f7150574431a59c7f)

Reply:
Sounds less clear to me tbh. What this api is supposed to provide users with is a projection of a price for a mint or a burn of a given size. There's a word for a price estimate for a transaction in the English language, and it's _quote_. (NOT _quota_ though, that's a different thing!). 'Get collateral' can be everything and anything. The currency in which collateral is provided, the total value locked in collateral, the total amount of collateral given a certain supply of bonded tokens, etc.